mainnet/gateways/lua.mjs

```diff
@@ -0,0 +1,134 @@
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import fs from 'fs';
+import path from 'path';
+
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const argv = yargs(hideBin(process.argv))
+  .option('dryRun', {
+    alias: 'd',
+    type: 'boolean',
+    description: 'Run without making any changes',
+    default: false,
+  })
+  .option('output', {
+    alias: 'o',
+    type: 'string',
+    description: 'Output CSV file path',
+    default: './outputs/gateways.lua',
+  })
+  .option('input', {
+    alias: 'i',
+    type: 'string',
+    description: 'Input CSV file path',
+    default: './outputs/gateways.csv',
+  })
+  .option('startTimestamp', {
+    alias: 's',
+    type: 'number',
+    description: 'Start timestamp (defaults to 2025-02-20 00:00:00 UTC)',
+    default: 1740009600000,
+  })
+  .help()
+  .parseSync();
+
+const dryRun = argv.dryRun;
+const output = path.join(__dirname, argv.output);
+const input = path.join(__dirname, argv.input);
+const startTimestamp = argv.startTimestamp; // the gateways starting timestamp
+console.log(
+  'Creating raw lua gateways with assigned delegates from',
+  input,
+  'and writing to',
+  output,
+);
+
+// mkdir if not exists
+fs.mkdirSync(path.dirname(output), { recursive: true });
+
+// overwrite the file if it exists
+fs.writeFileSync(output, '');
+
+// read all the records from the input file
+// gatewayAddress,observerAddress,operatorStake,fqdn,port,protocol,allowDelegatedStaking,delegateRewardShareRatio,minDelegatedStake,autoStake,label,note,properties,status,failedConsecutiveEpochsCount
+const gateways = fs
+  .readFileSync(input, 'utf8')
+  .split('\n')
+  .slice(1)
+  .map((row) => row.trim())
+  .filter((row) => row.length > 0)
+  .filter(Boolean) // Remove empty lines
+  .map((row) => row.split(','));
+
+for (const gateway of gateways) {
+  const [
+    gatewayAddress,
+    observerAddress,
+    operatorStake,
+    fqdn,
+    port,
+    protocol,
+    allowDelegatedStaking,
+    delegateRewardShareRatio,
+    minDelegatedStake,
+    autoStake,
+    label,
+    note,
+    properties = 'FH1aVetOoulPGqgYukj0VE0wIhDy90WiQoV3U2PeY44',
+    status,
+    failedConsecutiveEpochsCount,
+  ] = gateway;
+  const luaRecord = `
+GatewayRegistry["${gatewayAddress}"] = {
```

Comment:
Can you explain the process you've used to ensure that every critical value is set on the output objects? Are we primary relying on automated tests to catch issues?

Reply:
we will continue to iterate on these scripts - they are non-blocking from this PR

Reply:
yes - i validate and use our monitor tests to validate every gateway,record, vault has the expected timestamp. the SDK also has suites of e2e tests to validate response types from the processes created with this injected state.